webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/reports/StatusReports.scala

```diff
@@ -278,11 +280,60 @@ object DirectiveStatusReport {
  * For a component, store the report status, as the worse status of the component
  * Or error if there is an unexpected component value
  */
-final case class ComponentStatusReport(
+sealed trait ComponentStatusReport extends  StatusReport {
+  def componentName : String
+  def getValues(predicate: ComponentValueStatusReport => Boolean): Seq[(String, ComponentValueStatusReport)]
+  def withFilteredElement(predicate: ComponentValueStatusReport => Boolean): Option[ComponentStatusReport]
+  def componentValues : Map[String, ComponentValueStatusReport]
+  def status : ReportType
+}
+
+final case class BlockStatusReport (
+    componentName : String
+  , reportingLogic: ReportingLogic
+  , subComponents : List[ComponentStatusReport]
+) extends  ComponentStatusReport {
+  def findChildren(componentName : String): List[ComponentStatusReport] = {
+    subComponents.find(_.componentName == componentName).toList :::
+    subComponents.collect{case g :BlockStatusReport => g}.flatMap(_.findChildren(componentName))
+  }
+  def compliance: ComplianceLevel = {
+    import ReportingLogic._
+    reportingLogic match {
+      case WorstReport =>
+        ComplianceLevel.compute(ReportType.getWorseType(subComponents.map(_.status)) :: Nil)
+      case SumReport => ComplianceLevel.sum(subComponents.map(_.compliance))
+      case FocusReport(component) => ComplianceLevel.sum(findChildren(component).map(_.compliance))
```

Comment:
I think that if the component is not found, it should not be 0 or whatever `ComplianceLevel.sum(Nil)` does, but it should be a missing report error.